framework/src/main/java/org/checkerframework/framework/util/CheckerMain.java

```diff
@@ -432,6 +450,29 @@ protected void addMainToArgs(final List<String> args) {
             args.add(quote(concatenatePaths(ppOpts)));
         }
 
+        if (PluginUtil.getJreVersion() > 8) {
+            // Get Patch_Module_argfile
+            final File patchModuleArgsFile = new File(jdkJar, "Patch_Modules_argfile");
+
+            // Get ${CHECKERFRAMEWORK}/checker/dist path
+            final String checkerframeworkDistDirectoryPath = checkerJar.getParentFile().getPath();
+
+            // adding "--patch-module" arguments to compiler Arguments
+            try {
+                for (final String argLine : PluginUtil.readFile(patchModuleArgsFile)) {
+                    String[] separateFlag = argLine.trim().split("\\s+");
+                    args.add(separateFlag[0]);
```

Comment:
How about you verify here that this is `--patch-module`? Or are there other valid options?

Reply:
Since we are going to generate `Patch_Module_argsfile`, I am not sure if we should check for its contents as it is only going to contain `--patch-module` options.

Reply:
I think it would be good to describe your assumptions about the `Patch_Module_argsfile` file and the content of that directory somewhere. The tool that will auto-generate that file will need to rely on that same specification. For example, the substitution of `CURRENT_JDK_FOLDER` should be explained somewhere.
Also, that file could be generated manually, so a user needs to know what the format is.

Reply:
Again, could you add such documentation somewhere? You now added a bit in `PluginUtil` below, but that does not seem like the best place for this information. Even a few inline comments here would be helpful.